src/control/server/server.go

```diff
@@ -110,10 +114,31 @@ func Start(log *logging.LeveledLogger, cfg *Configuration) error {
 		TargetUser:    runningUser.Username,
 		PCIWhitelist:  strings.Join(cfg.BdevInclude, ","),
 	}
+	log.Debugf("automatic NVMe prepare req: %+v", prepReq)
 	if _, err := bdevProvider.Prepare(prepReq); err != nil {
 		log.Errorf("automatic NVMe prepare failed (check configuration?)\n%s", err)
 	}
 
+	hugePages, err := getHugePageInfo()
+	if err != nil {
+		return errors.Wrap(err, "unable to read system hugepage info")
+	}
+
+	// Don't bother with these checks if there aren't any block devices configured.
+	if cfgHasBdev(cfg) {
```

Comment:
Should we skip the bdev Prepare call if there are no block devices? Or is that required before we even get the hugepage info?

Reply:
Yeah, we need to do the Prepare() before we check the hugePage info because that's what sets things up. Also, we have to do the Prepare() even without bdev devices in the config because that's what will allow the initial storage scan use case to work. The goal is to remove the requirement for the admin to explicitly run `storage prepare --nvme-only` before being able to figure out what kind of storage is available.